pytorch_lightning/utilities/data.py

```diff
@@ -336,23 +352,28 @@ def wrapper(obj: DataLoader, *args: Any, **kwargs: Any) -> None:
             if param.name != "self" and param.kind not in (param.VAR_POSITIONAL, param.VAR_KEYWORD)
         ]
 
-        cls = type(obj)
+        set_arg_names = []
         for arg_name, arg_value in chain(zip(param_names, args), kwargs.items()):
-            if hasattr(cls, arg_name) and getattr(cls, arg_name).fset is None:
-                # the class defines a read-only (no setter) property of this name. it's likely that the implementation
-                # will set `self._arg_name = arg_value` in `__init__` which is the attribute returned by the `arg_name`
-                # property so we are fine skipping in that case
-                continue
-            setattr(obj, arg_name, arg_value)
```

Comment:
Not setting this would now be a breaking change for people using 1.6 who rely on this behaviour

Reply:
Do people rely on this particular behavior? Or do they rely on the fact that their class will be reinstantiated correctly? I believe how we manage the reinstantiation is an implementation detail.

I believe a good "test" for whether people rely on this particular behavior and we shouldn't remove it is whether we advertised it in the docs (such as "if you create a DataLoader in a hook, you don't have to set your attributes, we'll do it for you")

Reply:
Somebody might (maybe PyTorch geometric?) but I agree their authors should create the attribute themselves.

Still I would mention this explicitly as a breaking change

Reply:
Mentioned it in breaking changes. Should I also update Changelog?

Reply:
Yes

Reply:
Done